src/libsyntax/parse/parser.rs

```diff
@@ -5102,6 +5103,29 @@ impl<'a> Parser<'a> {
                 let vis = Visibility::Restricted { path: P(path), id: ast::DUMMY_NODE_ID };
                 self.expect(&token::CloseDelim(token::Paren))?; // `)`
                 return Ok(vis)
+            } else if self.look_ahead(0, |t| *t == token::Token::OpenDelim(token::Paren)) {
```

Comment:
Looks like this is always true, we have already checked for `(` in the outer condition.